Project-5.md

```diff
@@ -46,9 +46,501 @@ You may find the following resources useful:
+
+] .
+
+Constraint for Reflexivity 
+a.	Our constraint checks for reflexivity in a graph.
+
+b.	An example of when one would use this constraint would be if we had a graph that represented published philosophers to check for when philosophers are self-referencing themselves in a novel piece. Such that when a philosopher published novel work that references that philosophers in a self-referential way this constraint would alert to the self-reference. One would change ex:ReflexivityShape to ex:PhilosopherReferenceReflexivityShape and follow the format of the constraint below. Importantly, when self-referring, the philosopher is referring to herself (rather than the body of work).  
+
+c.	a -> a
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+ex:ReflexiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 6 ;    
+
+sh:datatype xsd:string ; 
+
+] .
+
+5.	Constraint for Irreflexive 
+a.	Our constraint checks for irreflexively in a graph 
+
+b.	An example of when to use this constraint would be if you had a graph that represented a company’s organizational structure. Such that no employee should be their own manager. This constraint would ensure that is not possible. One would simply change ex:IrrelexiveShape to ex:CompanyIrrelexiveOrganizationShape and follow the formatting below. 
+
+c.	~(a -> a)
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:IrreflexiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+
+sh:maxCount 4 ;    
+
+sh:datatype xsd:string ; 
+] .
+
+6.	Constraint for Transitivity 
+a.	Our constraint checks for transitivity in a graph.
+
+b.	An example of when one would use this constraint would be to check a graph of clan, band, nation affiliation. Such that when one is a recognized citizen of the Cherokee Nation one is also a member of one of the 3 bands of Cherokee and a member of one of the 7 clans. This constraint would be used to check if one is a member of a clan, they are a member of a band and if you are member a band you are a member of the Nation and if you are a member of a clan, you are a member of the Nation. One would simply change ex:TransitiveShape to ex:CherokeeNationEnrollmentClanTransitiveShape.
+
+c.	c -> b. b -> a. ؞ c -> a
+  i.	a = nation
+  ii.	b = band
+  iii.	c = clan
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:TransitiveShape
```

Comment:
This shape is just the preceding shape with a different maxcount... 

Even if you'd modeled irreflexivity correctly in the preceding shape, why would changing the maxcount model transitivity?